src/jit/lowerarmarch.cpp

```diff
@@ -382,6 +382,11 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
         if (source->gtOper == GT_IND)
         {
             MakeSrcContained(blkNode, source);
+            GenTree* addr = source->AsIndir()->Addr();
+            if (!addr->OperIsLocalAddr())
+            {
+                addr->ClearContained();
```

Comment:
Does `MakeSrcContained` set contained for the `addr'? Why can't it check that in this case it should not set it?

Reply:
`MakeSrcContained` only sets `source` as contained. It doesn't check any of the children of `source` - and in some cases it is perfectly valid for an indir to be contained as well as a child `LEA`.

Reply:
Also note that this same code was already here: https://github.com/dotnet/coreclr/blob/master/src/jit/lowerxarch.cpp#L383